http-clients/java-nio-client/src/test/java/software/amazon/awssdk/http/nio/java/internal/S3JavaHttpClientIntegratedTest.java

```diff
@@ -0,0 +1,159 @@
+package software.amazon.awssdk.http.nio.java.internal;
+
+
+import static junit.framework.TestCase.assertEquals;
+import static org.apache.commons.lang3.RandomStringUtils.randomAlphabetic;
+import java.util.concurrent.CompletableFuture;
+import org.junit.Test;
+import software.amazon.awssdk.core.ResponseBytes;
+import software.amazon.awssdk.core.async.AsyncRequestBody;
+import software.amazon.awssdk.core.async.AsyncResponseTransformer;
+import software.amazon.awssdk.core.retry.RetryPolicy;
+import software.amazon.awssdk.http.Protocol;
+import software.amazon.awssdk.http.async.SdkAsyncHttpClient;
+import software.amazon.awssdk.http.nio.java.JavaHttpClientNioAsyncHttpClient;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3AsyncClient;
+import software.amazon.awssdk.services.s3.S3Configuration;
+import software.amazon.awssdk.services.s3.model.*;
+
+public class S3JavaHttpClientIntegratedTest {
+
+    private static final String BUCKET = "zhouquas3testbucketjava";
+    private static final String KEY = "smalltestkey";
+    private static final String LARGEKEY = "largetestkey";
+    private static final String VALUE = "Hello World!";
+    private static String LARGESTRING = randomAlphabetic(25000);
+
+    private S3AsyncClient s3Async;
+
+    private void createS3AsyncClient(SdkAsyncHttpClient myClient) {
+        this.s3Async = S3AsyncClient.builder()
+                .httpClient(myClient)
+                .overrideConfiguration(b -> b.retryPolicy(RetryPolicy.none()))
+                .serviceConfiguration(S3Configuration.builder().checksumValidationEnabled(false).build())
+                .region(Region.US_WEST_2)
+                .build();
+    }
+
+    private void createJavaHttpClientHTTP1() {
+        SdkAsyncHttpClient javaHttpCient = JavaHttpClientNioAsyncHttpClient.builder().protocol(Protocol.HTTP1_1).build();
+        createS3AsyncClient(javaHttpCient);
+    }
+
+    private void createJavaHttpClientHTTP2() {
+        SdkAsyncHttpClient javaHttpCient = JavaHttpClientNioAsyncHttpClient.builder().protocol(Protocol.HTTP2).build();
+        createS3AsyncClient(javaHttpCient);
+    }
+
+    @Test
+    public void s3creatBucketWithJavaHttpClientHTTP1Test() {
+        createJavaHttpClientHTTP1();
+        createBucketTest(s3Async);
```

Comment:
Does this test work? Creating a bucket that already exists should throw an exception

Reply:
A try catch clause has been added.